component.js

```diff
@@ -95,19 +95,21 @@ module.exports.withDefaults = factory;
 function factory (options) {
   var debug;
   options = options || {};
-  var _shouldComponentUpdate = options.shouldComponentUpdate;
+  var _componentWillMount = options.componentWillMount ||
+                            componentWillMount;
+  var _shouldComponentUpdate = options.shouldComponentUpdate ||
+                               shouldComponentUpdate.withDefaults(options);
+  var _componentWillReceiveProps = options.componentWillReceiveProps ||
```

Comment:
Yeah, we should add our mixin first, but I don't know if React guarantee the order in which they are invoked. Trying to find documentation. However, as it is a list, the order should be retained.